website/www/site/content/en/documentation/sdks/typescript.md

```diff
@@ -0,0 +1,146 @@
+[problematic Pipeline object](https://s.apache.org/no-beam-pipeline)
+from the API, instead providing a `Root` PValue on which pipelines are built,
+and invoking run() on a Runner.  We offer a less error-prone `Runner.run`
+which finishes only when the pipeline is completely finished as well as
+`Runner.runAsync` which returns a handle to the running pipeline.
+
+* Rather than introduce PCollectionTuple, PCollectionList, etc. we let PValue
+literally be an
+[array or object with PValue values](https://github.com/robertwb/beam-javascript/blob/de4390dd767f046903ac23fead5db333290462db/sdks/node-ts/src/apache_beam/pvalue.ts#L116)
+which transforms can consume or produce.
+These are applied by wrapping them with the `P` operator, e.g.
+`P([pc1, pc2, pc3]).apply(new Flatten())`.
+
+* Like Python, `flatMap` and `ParDo.process` return multiple elements by
+yielding them from a generator, rather than invoking a passed-in callback.
+There is currently an operation to split a PCollection into multiple
+PCollections based on the properties of the elements, and
+we may consider using a callback for side outputs.
+
+* The `map`, `flatMap`, and `ParDo.process` methods take an additional
+(optional) context argument, which is similar to the keyword arguments
+used in Python. These are javascript objects whose members may be constants
+(which are passed as is) or special DoFnParam objects which provide getters to
+element-specific information (such as the current timestamp, window,
+or side input) at runtime.
+
+* Rather than introduce multiple-output complexity into the map/do operations
+themselves, producing multiple outputs is done by following with a new
+`Split` primitive that takes a
+`PCollection<{a?: AType, b: BType, ... }>` and produces an object
+`{a: PCollection<AType>, b: PCollection<BType>, ...}`.
+
+* JavaScript supports (and encourages) an asynchronous programing model, with
+many libraries requiring use of the async/await paradigm.
+As there is no way (by design) to go from the asynchronous style back to
+the synchronous style, this needs to be taken into account
+when designing the API.
+We currently offer asynchronous variants of `PValue.apply(...)` (in addition
+to the synchronous ones, as they are easier to chain) as well as making
+`Runner.run` asynchronous. TBD to do this for all user callbacks as well.
+
+An example pipeline can be found at [wordcount.ts](https://github.com/apache/beam/blob/master/sdks/typescript/src/apache_beam/examples/wordcount.ts)
+and more documentation can be found in the [beam programming guide](/documentation/programming-guide/).
+
+
+## Pipeline I/O
+
+See the [Beam-provided I/O Transforms](/documentation/io/built-in/) page for a list of the currently available I/O transforms.
+
+
+## Supported Features
+
+The Typescript SDK is still under development but already supports many,
+but not all, features currently supported by the Beam model, including those required by streaming.
```

Comment:
"including those required by streaming." - it is ambiguous whether this means these features are or are not included. I think you're saying they're not included since we don't have fine grained watermark-type stuff, but I also think you could write a streaming pipeline using x-lang, so I'm not sure if the specific callout makes sense.

Reply:
It actually does handle all the streaming stuff except state and timers. Reworded. (A different runner is required for some of the features though.)